src/methods.ts

```diff
@@ -254,6 +347,11 @@ export class Method extends AbstractMethodField {
    */
   private code: any;
 
+  private numOpcodeExecs = 0;
+  private jitThreshold = 0;
+  private compiledFunctions: Function[] = [];
```

Comment:
I am not sure I understood this. Are you suggesting that we cache the JIT emitted function's source, instead of caching the compiled function?